docs/cep/windows/window-types/tumbling-expression.md

```diff
@@ -0,0 +1,31 @@
+---
+title: TUMBLING_EXPRESSION()
+---
+
+A batch (tumbling) length window that holds and process a number of events as specified in the `length`.
+
+## Syntax
+
+    WINDOW TUMBLING_EXPRESSION(expression <STR>)
+    WINDOW TUMBLING_EXPRESSION(expression <STR>, current.event <BOOL>)
+
+## Query Parameters
+
+| Name      | Description       | Default Value | Possible Data Types | Optional | Dynamic |
+|-----------|---------------------|------------|--------------|----------|---------|
+| expression        | An expression that defines what events the window should tumble. All possible expressions are allowed.         |      | STR  | No       | No      |
+| current.event | Let the window stream the current events out as and when they arrive to the window while expiring them in batches. | false | BOOL    | Yes      | No      |
+
+## Example 1
+
+```sql
+CREATE STREAM InputEventStream (symbol string, price float, volume int);
```

Comment:
@koshy-thomas and @StoyanTc, this stream worker is invalid, probably because of the expression in the window definition. I didn't know what to put there.

Please give this whole topic a thorough look, as it is brand new content.

Reply:
@diana-macrometa you can use this as example:

CREATE STREAM cseEventStream (symbol string, price float, volume int);
CREATE WINDOW cseEventWindow (symbol string, price float, volume int) TUMBLING_EXPRESSION('count() < 5');
CREATE SINK OutputStream WITH (type='stream', stream='OutputStream') (symbol string, price double);

@info(name = 'query0')
INSERT INTO cseEventWindow
FROM cseEventStream;

@info(name = 'query1')
INSERT INTO OutputStream
SELECT symbol, sum(price) AS price
FROM cseEventWindow;


It will give output only after inserting 5 valid messages into the cseEventStream.
Let me know if you have doubts
Also you can add the valid keyword here for expressions..

Reply:
Already added "valid" to both expression topics.